### PR TITLE
fix(register): update error message when receiving improper error

### PR DESCRIPTION
**Pull request by codelemur**
I decided to try and verify that the error.message we receive is actually a string, otherwise to
just render the default error

So there are actually quite a few situations in which the API responds with an object as the error message, but only a few places we try to render that to the UI. In this instance, I think the better solution was to check the typing that comes back from the API and to render the default if there's anything other than a `string` in the `error.message`. I think we could refactor the API if we wanted to, but for this particular issue (TDX-1687), you'd have to change a bunch of endpoints on the API side.

Here's what it'll look like (devtool screenshot too)
image

### CODE REVIEW

File: workspaces/default/themes/base/partials/auth/forms/register/basic-auth.html [hunk 1]
```diff
@@ -76,9 +76,10 @@
               .text(response.fields[field])
           }
         } else {
+          const uiErrorMessage = data && data.message && typeof data.message  === "string" ? data.message : fallbackErrorMessage;
```
Comment (Mierenga) [inline workspaces/default/themes/base/partials/auth/forms/register/basic-auth.html:79]: I like the improvement of ensuring the message is a string before rendering it so we don't ever see `[Object object]`. The response in the screenshot shows we have some information from this error that we can render to help the user out a little more. What do you think about a function that works like this:

```
function getErrorMessage(error) {
  const defaultMessage = 'An unexpected error occurred. Please try again.'

  const isObject = o => o && typeof o === 'object'

  if (!error) { return defaultMessage }

  if (error.message && typeof error.message === 'string') { return error.message }

  if (isObject(error.message)) {
    if (isObject(error.message.error)) {

      // Get message for errors with problems sending emails:
      // {
      //   message: {
      //     error: { emails: { 'noreply@konghq.com': 'Error sending email'}, count: 1 },
      //     sent: { emails: {}, count: 0 }
      //   }
      // }
      if (isObject(error.message.error.emails)) {
        return Object.entries(error.message.error.emails).map(([k, v]) => {
          return `${v} from ${k}`
        }).join('\n')
      }
    }
  }

  return defaultMessage
}
```
e.g.,
```
const message = getErrorMessage({
  message: {
    error: { emails: { 'noreply@konghq.com': 'Error sending email'}, count: 1 },
    sent: { emails: {}, count: 0 }
  }
})

// Should output: 'Error sending email from noreply@konghq.com\n'
console.log(message)
```

Then we can add more cases to the `getErrorMessage()` function in the future if we want to create more friendly messages for other things.

**Comment by codelemur** [comment]
Alright updated to be more specific and provide a handler for future extensibility:
image

File: workspaces/default/themes/base/assets/js/kong/kong.utils.js [hunk 2]
```diff
@@ -1,46 +1,61 @@
...
+    // we want errors to be unique
+    let foundErrors = new Set();
+    // Handle SMTP sending errors:
+    for (const email in response.message.error.emails) {
+      const errorMessage = response.message.error.emails[email];
+      foundErrors.add(errorMessage);
+    }
+    return Array.from(foundErrors).join("\n");
+  } catch (error) {
+    return fallbackErrorMessage;
```
Comment (Mierenga) [inline workspaces/default/themes/base/assets/js/kong/kong.utils.js:59]: Are we missing one case where we would want to use `response.message` itself if it is a string?

Comment (codelemur) [reply workspaces/default/themes/base/assets/js/kong/kong.utils.js:64]: :doh: thanks

**Review by Mierenga — APPROVED** [review]
Tested locally with an invalid SMTP configuration, looks good!

**Pull request merged**